nntrainer/tensor/q4_k_tensor.cpp

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * @file	q4_k_tensor.cpp
+ * @date	23 April 2025
+ * @brief	This is Q4_K_Tensor class for Q4_K quantized tensor.
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Donghyeon Jeong <dhyeon.jeong@samsung.com>
+ * @bug		No known bugs except for NYI items
+ */
+
+#include <cpu_backend.h>
+#include <q4_k_tensor.h>
+#include <tensor.h>
+
+namespace nntrainer {
+
+Q4_K_Tensor::Q4_K_Tensor(std::string name_, Tformat fm, QScheme qscheme_) :
+  Uint4QTensor(name_, fm, QScheme::Q4_Kx8) {}
+
+Q4_K_Tensor::Q4_K_Tensor(const TensorDim &d, bool alloc_now, Initializer init,
+                         std::string name, QScheme qscheme_) :
+  Uint4QTensor(d, alloc_now, init, name, qscheme_) {
+  NNTR_THROW_IF(d.batch() != 1 || d.channel() != 1 ||
+                  (d.height() * d.width()) % 256 != 0,
+                std::invalid_argument)
+    << "Q4_K_Tensor must be 2 dimensional tensor with batch size 1 and "
+       "height * width must be divisible by 256";
```

Comment:
Think I made you confused a little bit:
```suggestion
                  (d.height() % 256 != 0 || d.width()% 256 != 0),
                std::invalid_argument)
    << "Q4_K_Tensor must be 2 dimensional tensor with batch size 1 and "
       "height * width must be divisible by 256";
```

Reply:
thanks!

Reply:
To clarify, is there one block_q4_Kx8 for each 256 elements or 256 x 256 elements?

Reply:
Hope this solves your question:

- 256 float values - q4_K quantization -> 1 block_q4_K
- 8 block_q4_k - repack -> block_q4_Kx8

Thus one block_q4_Kx8 represent 8*256 original float values, AFAIK

Reply:
Thanks for the clarification! I get it now. However, I don't understand why both height and width should be divisible by 256. If one block of q4x8 represents 2048 float values, shouldn't the tensor size be divisible by 2048?

Reply:
Thanks for the clarification!